private/buf/cmd/buf/command/lint/lint.go

```diff
@@ -175,3 +228,38 @@ func run(
 	}
 	return nil
 }
+
+func filterNewFileAnnotations(
+	allFileAnnotations []bufanalysis.FileAnnotation,
+	diffFromRevision, diffPatchFilePath string,
+	onlyNew, wholeFiles bool,
+) ([]bufanalysis.FileAnnotation, error) {
+	if !onlyNew && diffFromRevision == "" && diffPatchFilePath == "" {
+		return allFileAnnotations, nil
+	}
+
+	var patchReader io.Reader
+	if diffPatchFilePath != "" {
+		patch, err := os.ReadFile(diffPatchFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("can't read from patch file %s: %s", diffPatchFilePath, err)
+		}
+		patchReader = bytes.NewReader(patch)
+	}
+	checker := revgrep.Checker{
```

Comment:
`revgrep` seems to only work with a local directory, whereas `buf` works with local directories (git or not git), tarballs, zip files, remote git repositories, and images. I don't think `revgrep` works as a general solution for `buf` here unless I'm missing something